tests/unit/test_charm.py

```diff
@@ -453,15 +459,14 @@ def test_create_snapshot_action_strip_path(
             ),
         )
 
-    @patch("os.walk")
-    def test_list_snapshots_action(self, os_walk):
-        def a2g(x):
-            return ([n, ["snapshot-{}".format(n)]] for n in x)
-
+    def test_list_snapshots_action(self):
         rand_n = random.randint(10, 100)
-        os_walk.return_value = a2g([rand_n])
         action_event = Mock()
         self.harness.charm._get_snapshot_name = Mock()
+        snapshot = Mock()
+        snapshot.name = "snapshot-{}".format(rand_n)
+        self.harness.charm._list_snapshots = Mock()
+        self.harness.charm._list_snapshots.return_value = [snapshot]
         self.harness.charm._on_list_snapshots_action(action_event)
         self.assertEqual(
```

Comment:
I did not change to `assert_called_once_with` since it was originally how is written, I think changing to `assert_called_once_with` is also okay. 

Agree, we should also test that because it is also not tested before.